share/ipld/get.go

```diff
@@ -170,22 +179,25 @@ func GetLeaves(ctx context.Context,
 	wg.Wait()
 }
 
-// GetLeavesByNamespace returns as many leaves from the given root with the given namespace.ID as
-// it can retrieve. If no shares are found, it returns both data and error as nil.
-// A non-nil error means that only partial data is returned, because at least one share retrieval
-// failed The following implementation is based on `GetShares`.
+// GetLeavesByNamespace returns leaves and corresponding proof that could be used to verify leaves
+// inclusion. It as many leaves from the given root with the given namespace.ID as it can retrieve.
+// If no shares are found, it returns both data and error as nil. collectProofs param indicates
+// whether proofs should be collected for shares inclusion verification. A non-nil error means that
+// only partial data is returned, because at least one share retrieval failed The following
+// implementation is based on `GetShares`.
 func GetLeavesByNamespace(
 	ctx context.Context,
 	bGetter blockservice.BlockGetter,
 	root cid.Cid,
 	nID namespace.ID,
 	maxShares int,
-) ([]ipld.Node, error) {
+	collectProofs bool,
+) (*LeavesWithProofs, error) {
```

Comment:
api suggestion:
```go
proofs := Proofs{} // create proofs struct
GetLeavesByNamespace(..., WithProofs(proofs)) // supply it as an option
// use the populated proofs struct
```

Reply:
I would really love to simplify the API to something like this. Do we think it is worth the time?

Reply:
The fact that we return the `LeavesWithProofs` struct always, even when proofs are not requested makes me feel bad

Reply:
Ok, we decided to avoid options for now and just to pass Proofs struct as the last params. If you don't need proofs, pass nil